object_storage_api/repositories/image.py

```diff
@@ -49,13 +50,21 @@ def get(self, image_id: str, session: ClientSession = None) -> Optional[ImageOut
         :param image_id: ID of the image to retrieve.
         :param session: PyMongo ClientSession to use for database operations.
         :return: Retrieved image or `None` if not found.
+        :raises MissingRecordError: If the supplied image_id is invalid or non-existent.
         """
-        image_id = CustomObjectId(image_id)
         logger.info("Retrieving image with ID: %s from the database", image_id)
-        image = self._images_collection.find_one({"_id": image_id}, session=session)
+        try:
+            image_id = CustomObjectId(image_id)
+            image = self._images_collection.find_one({"_id": image_id}, session=session)
+        except InvalidObjectIdError as e:
+            exc = MissingRecordError(f"The image_id {image_id} is not valid.")
+            exc.response_detail = f"Invalid image_id format: {image_id}"
+            raise exc from e
```

Comment:
```suggestion
        except InvalidObjectIdError as e:
            exc = MissingRecordError(f"The image_id {image_id} is not valid.")
            exc.response_detail = "Image not found"
            raise exc from e
```
I would go with `Image not found` as this is like Viktor requested, thanks to the raise exc from e - it logs the earlier one too specifying its due to an invalid id.

I would also modify BaseAPIException's init to take in an optional response detail that assigns it if it isn't None, so it can be put on one line e.g. `raise MissingRecordError(detail, response_detail) from exc`. You could also override the init in `MissingRecordError` to take in the `entity_name` and automatically assigning the response detail to `f"{entity_name.capitalize()} was not found"` to make it even simpler.